llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -2648,3 +2688,112 @@ Expected<DIEDataRef> DIEDataRef::create(MCCASBuilder &MB,
   append_range(B->Data, toStringRef(Contents));
   return get(B->build());
 }
+
+// Returns true if DIE should be placed in a separate CAS block.
+static bool shouldCreateSeparateBlockFor(DWARFDie &DIE) {
+  dwarf::Tag Tag = DIE.getTag();
+  if (!is_contained({dwarf::Tag::DW_TAG_subprogram}, Tag))
+    return false;
+
+  for (const DWARFAttribute &AttrValue : DIE.attributes())
+    if (AttrValue.Attr == dwarf::Attribute::DW_AT_low_pc)
```

Comment:
A comment on why you are splitting on DW_AT_low_pc would be nice!

Reply:
Yup, good point!